airbyte-integrations/connectors/source-airtable/source_airtable/source.py

```diff
@@ -10,21 +10,20 @@
 from airbyte_cdk.models import AirbyteCatalog
 from airbyte_cdk.sources import AbstractSource
 from airbyte_cdk.sources.streams import Stream
-from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
 
+from .auth import AirtableAuth
 from .schema_helpers import SchemaHelpers
 from .streams import AirtableBases, AirtableStream, AirtableTables
 
 
 class SourceAirtable(AbstractSource):
 
     logger: logging.Logger = logging.getLogger("airbyte")
-
-    # prepared streams catalog
     streams_catalog: Iterable[Mapping[str, Any]] = []
+    _auth: AirtableAuth = None
```

Comment:
I see to the class variable `_auth` is used by 2 diff functions: `streams` and `check_connection`. I am not sure to these ones can be called by same python script call and this code optimisation will have some benefit because AirByte engine calls them by diff commands. Please correct me if I am not right.

Reply:
You're correct, therefore in order to have a local test and development more manageable, we need to save the auth for later use, if possible. The Platform calls the check, discover, and streams using different commands; therefore, while using `SingleUseRefreshToken` we need to be sure that the same `access_token` acquired from OAuth2.0 will be passed to all the stream class instances and will not be triggered for each stream generated.